src/test/java/org/opensearch/sdk/TestSDKRestRequest.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+import org.opensearch.common.bytes.BytesArray;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.extensions.rest.ExtensionRestRequest;
+import org.opensearch.http.HttpRequest;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestRequest.Method;
+import org.opensearch.sdk.rest.SDKHttpRequest;
+import org.opensearch.sdk.rest.SDKRestRequest;
+import org.opensearch.test.OpenSearchTestCase;
+
+import static java.util.Map.entry;
+
+public class TestSDKRestRequest extends OpenSearchTestCase {
+    @Test
+    public void TestSDKRestRequestMethods() {
+        RestRequest.Method expectedMethod = Method.GET;
+        String expectedUri = "foobar?foo=bar&baz=42";
+        String expectedPath = "foo";
+        Map<String, String> expectedParams = Map.ofEntries(entry("foo", "bar"), entry("baz", "42"));
+        Map<String, List<String>> expectedHeaders = Map.ofEntries(entry("foo", Arrays.asList("hello", "world")));
+        XContentType exptectedXContentType = null;
+        BytesReference expectedContent = new BytesArray("bar");
```

Comment:
If we pass actual JSON here (`"{\"foo\":\"bar\"}`" then we can actually test the extracted result of the `contentParser()` method from the RestRequest:
```java
Map<String, Object> source = request.contentParser().map();
assertEquals("bar", (String) source.get("foo")); // I think?
```